src/update_validation.rs

```diff
@@ -19,17 +17,17 @@ static UPDATE_VALIDATION_FILE: &str = "/run/omnect-device-service/omnect_validat
 static UPDATE_VALIDATION_COMPLETE_BARRIER_FILE: &str =
     "/run/omnect-device-service/omnect_validate_update_complete_barrier";
 static IOT_HUB_DEVICE_UPDATE_SERVICE: &str = "deviceupdate-agent.service";
-static UPDATE_VALIDATION_TIMEOUT_IN_SECS: u128 = 300;
+static UPDATE_VALIDATION_TIMEOUT_IN_SECS: u64 = 300;
 
 #[derive(Default, Deserialize, Serialize)]
 pub struct UpdateValidation {
-    start_monotonic_time_ms: u128,
+    start_monotonic_time: Duration,
     restart_count: u8,
     authenticated: bool,
     #[serde(skip)]
     run_update_validation: bool,
     #[serde(skip)]
-    validation_timeout_ms: u128,
+    validation_timeout: Duration,
```

Comment:
was this tested against `omnect-device-service.exec_stop_post.sh`? is the serialization of a `Duration` the same as u128?

i guess no, because `omnect-device-service.exec_stop_post.sh` is not adapted, and you changed the interface (struct vs json).

Reply:
Ok, we don't have a unit test which checks reading/writing the file. Maybe we should add one...
Anyway fixed....

Reply:
the scripts works with `start_monotonic_time_ms` not `start_monotonic_time`. 
you'll have to  adapt https://github.com/omnect/omnect-device-service/blob/b7d3a4184ea45ae4f510da345c5fe0a42f73a113/systemd/omnect-device-service.exec_stop_post.sh#L23

Reply:
stupid me. Didn't get that the discussion is just about the wrong var name.
fixed.